internal/handlers/common/aggregations/stages/projection/projection.go

```diff
@@ -232,6 +248,22 @@ func ProjectDocument(doc, projection *types.Document, inclusion bool) (*types.Do
 			var op operators.Operator
 			var value any
 
+			if !operators.IsOperator(idValue) {
+				if idValue.Len() == 0 {
+					return nil, commonerrors.NewCommandErrorMsgWithArgument(
+						commonerrors.ErrEmptySubProject,
+						"Invalid $project :: caused by :: An empty sub-projection is not a valid value."+
+							" Found empty object at path",
+						"$project (stage)",
+					)
+				}
```

Comment:
Great catch finding this. Could you also test this please?

Reply:
It's not possible as it's catched on validation.